pkg/kvstore/allocator/allocator_test.go

```diff
@@ -607,3 +612,74 @@ func TestRemoteCache(t *testing.T) {
 		require.Equal(t, 2, cache[i])
 	}
 }
+
+func TestRunGC(t *testing.T) {
+	testutils.IntegrationTest(t)
+
+	const (
+		minID = 0
+		maxID = 1000
+	)
+
+	var (
+		ctx    = t.Context()
+		log    = hivetest.Logger(t)
+		client = kvstore.SetupDummyWithConfigOpts(t, "etcd", etcdOpts)
+		base   = randomTestName()
+		rl     = rate.NewLimiter(1*time.Millisecond, 1000)
+	)
+
+	backend, err := NewKVStoreBackend(log, KVStoreBackendConfiguration{base, "", TestAllocatorKey(""), client})
+	require.NoError(t, err, "NewKVStoreBackend")
+
+	for id, label := range map[string]string{
+		"100": "k8s:foo=bar;",
+		"101": "k8s:foo=bar;baz=qux;",
+		"102": "k8s:foo=bar;fred=true;",
+	} {
+		// Create a primary entry for all identities
+		require.NoError(t, client.Update(ctx, path.Join(base, "id", id), []byte(label), false), "client.Update")
+
+		// Create two secondary entries for all identities
+		for _, node := range []string{"node-a", "node-b"} {
+			require.NoError(t, client.Update(ctx, path.Join(base, "value", label, node), []byte(id), false), "client.Update")
+		}
+	}
+
+	var (
+		trim = func(in iter.Seq[string]) iter.Seq[string] {
+			return cslices.MapIter(in, func(in string) string { return strings.TrimPrefix(in, path.Join(base, "id")+"/") })
+		}
```

Comment:
The `cslices` package doesn't exist in v1.18, so you'd need to open-code this function:

```go
		trim = func(in iter.Seq[string]) iter.Seq[string] {
			return func(yield func(string) bool) {
				for pfx := range in {
					if !yield(strings.TrimPrefix(pfx, path.Join(base, "id")+"/")) {
						return
					}
				}
			}
		}
```